ext/dba/dba_lmdb.c

```diff
@@ -40,30 +40,20 @@ DBA_OPEN_FUNC(lmdb)
 {
 	MDB_env *env;
 	MDB_txn *txn;
-	int rc, mode = 0644, flags = MDB_NOSUBDIR;
-	zend_long mapsize = 0;
-
-	if(info->argc > 0) {
-		mode = zval_get_long(&info->argv[0]);
-
-		if (info->argc > 1) {
-			mapsize = zval_get_long(&info->argv[1]);
-			if (mapsize < 0) {
-				*error = "mapsize must be greater than or equal to zero";
-				return FAILURE;
-			}
-		}
-		/* TODO implement handling of the additional flags. */
-	}
+	int rc, flags = MDB_NOSUBDIR;
+	int mode = info->file_permission;
+	zend_long map_size = info->map_size;
+
+	ZEND_ASSERT(map_size > 0);
```

Comment:
Is this assertion correct? Shouldn't it be `>= 0`?

Reply:
Yes. `0` uses the default mapsize which had been configured for lmdb.

Reply:
Indeed